app/controllers/search_controller.rb

```diff
@@ -11,4 +11,14 @@ def extract_filters
     respond_with TextSearchClient.parse_text(params[:q])
   end
 
+  def log_metrics_event
+    if params[:type] && params[:data] then
```

Comment:
Validate all the params here. (the gotomeeting issue)

Reply:
Beyond checking that they exist, what sort of validation are you looking for?

Reply:
No need to check the existence, but validate _all_ the possible values that will be passed in to this param, using ```==``` or ```.eql?```. For example, if params[:type] can have values of 'a', 'b', or 'c'. Then in this controller action, you should have something in place like this:
```if params[:type] == 'a'
  # code
elsif params[:type] == 'b'
  # code
elsif params[:type] == 'c'
  # code
else
  # code```

Reply:
We can talk about this in another gotomeeting.

Reply:
I'll have it check that "if params[:type] == 'reverb_redirect'" = right now there's only one 'type' coming from the frontend and that is it.

I think this function might be useful in the future for logging other front end events; when doing so though any new 'types' will have to be added to the conditional (minor programmatic overhead).